### PR TITLE
feat(onesync): ped movement group research

### PR DESCRIPTION
**Pull request by MRV6**
This PR adds 3 natives : 
- `GET_PED_STEALTH_MOVEMENT`
- `IS_PED_STRAFING`
- `IS_PED_RAGDOLL`

Tested across all game builds.

### CODE REVIEW

File: code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp [hunk 1]
```diff
@@ -1657,6 +1657,27 @@ static void Init()
...
+		auto movementGroup = entity->syncTree->GetPedMovementGroup();
+
+		return movementGroup ? movementGroup->isStealthy : false;
+	}));
+
+	fx::ScriptEngine::RegisterNativeHandler("IS_PED_FREE_AIMING", makeEntityFunction([](fx::ScriptContext& context, const fx::sync::SyncEntityPtr& entity)
+	{
+		auto movementGroup = entity->syncTree->GetPedMovementGroup();
+
+		return movementGroup ? movementGroup->isStrafing : false;
```
Comment (Disquse) [inline code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp:1672]: Are you sure that `isStrafing` here makes sense for `IS_PED_FREE_AIMING` native?

**Review by Disquse — COMMENTED** [review]
I guess `isStrafing` is a simple typo? Works fine in game however.

File: code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp (hunk 1 above)
Comment (MRV6) [reply code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp:1672]: That’s the name that I found using Xbox dump, but yes if needed I could change it

Comment (Disquse) [reply code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp:1672]: Hmm this is weird, are you sure this variable is not used for anything else instead? Like strafing or so?

Comment (MRV6) [reply code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp:1672]: Changed native name to `IS_PED_STRAFING` after some tests.

**Comment by blattersturm** [comment]
Comes with conflicts preapplied, nice! :(

**Comment by MRV6** [comment]
I have another PR #1217 opened which edits the same files so it will probably cause conflicts anyway.

**Comment by MRV6** [comment]
Fixed conflicts as #1217 has been merged since.

**Review by Disquse — APPROVED** [review]
Since conflicts was resolved. Marked as "ready-to-merge".

**Pull request closed without merging**